src/backend/dx12/src/device.rs

```diff
@@ -966,37 +963,43 @@ impl d::Device<B> for Device {
             });
     }
 
-    fn read_mapping_raw(
-        &mut self,
-        _buf: &n::Buffer,
-        _range: Range<u64>,
-    ) -> Result<(*const u8, Mapping), mapping::Error> {
-        unimplemented!()
-    }
-
-    fn write_mapping_raw(
-        &mut self,
-        buf: &n::Buffer,
-        range: Range<u64>,
-    ) -> Result<(*mut u8, Mapping), mapping::Error> {
-        if (range.end - range.start) > buf.size_in_bytes as _ {
-            return Err(mapping::Error::OutOfBounds);
-        }
-
-        let range = winapi::D3D12_RANGE {
-            Begin: range.start,
-            End: range.end,
+    fn acquire_mapping_raw(&mut self, buf: &n::Buffer, read: Option<Range<u64>>)
+        -> Result<*mut u8, mapping::Error>
+    {
+        let read_range = if let Some(read) = read {
```

Comment:
it's a matter of preference, but I somewhat like `match` more than `if let xxx=yyy {...} else {..}` because the latter is more sugar for nothing in return